src/components/Calculator.js

```diff
@@ -1,50 +1,178 @@
 import React, { Component } from 'react';
 import {
   TiDivide, TiTimes, TiMinus, TiPlus, TiEquals,
-} from 'react-icons/ti';
+}
+  from 'react-icons/ti';
+import calculate from '../mathLogic/calculate';
 
 class Calculator extends Component {
   constructor(props) {
     super(props);
-    this.state = {};
+    this.state = {
+      total: null,
+      next: null,
+      operation: null,
+    };
   }
 
+  handleCalculate = ({ currentTarget: btn }) => {
+    const btnName = btn.outerText === '' ? btn.id : btn.outerText;
+    const obj = calculate(this.state, btnName);
+    this.setState(obj);
+  };
+
   render() {
+    const { next, total } = this.state;
     return (
       <div className="container">
-        <div className="result">0</div>
+        <div className="result">
+          {total && next ? next : total || next || '0'}
+        </div>
         <ul className="cal-table">
-          <li className="cal-item">AC</li>
-          <li className="cal-item">+/-</li>
-          <li className="cal-item">%</li>
-          <li className="cal-item bg-orange">
+          <li
+            className="cal-item"
+            onClick={this.handleCalculate}
+            aria-hidden="true"
+          >
+            AC
+          </li>
+          <li
+            className="cal-item onClick={this.handleCalculate}"
+            onClick={this.handleCalculate}
+            aria-hidden="true"
+          >
+            +/-
+          </li>
+          <li
+            className="cal-item"
+            id="%"
+            onClick={this.handleCalculate}
+            aria-hidden="true"
+          >
+            %
+          </li>
```

Comment:
- [ ] Please double-check on this button's functionality because it doesn't seem to output the right results. 🙏 😄

Reply:
Hello, @MahmoudBakr23 I think that is not my mistake that is the way the function `calculate` that we copied should function. and I think I am not allowed to change the content of that function.

Reply:
KIndly everything is working well except the situation when we have an existing operation and you click on a new operation.